### PR TITLE
chore: Comply with the game's requested frame-rate.

### PR DESCRIPTION
**Pull request by ArjixWasTaken**
By having a manual frame-loop we can control (with some inaccuracies) the frame-rate.
I chose to use `setTimeout` instead of `setInterval` for two reasons.

1. `setInterval` doesn't wait for the previous iteration to end (allowing for duplicate frames)
2. due to the nature of `setInterval`, I'd have to call `clearInterval` anytime the `targetFPS` is changed.

### CODE REVIEW

**Comment by ArjixWasTaken** [comment]
I won't be making another PR to have the fps counter, since it is just 3 extra lines of code.

(and you know how to draw text using the DOM Canvas)

File: raylib.js [hunk 1]
```diff
@@ -96,17 +102,35 @@ WebAssembly.instantiateStreaming(fetch('wasm/game.wasm'), {
...
+
+    const frame = () => {
+        frameCount++;
+
+        const currentTime = performance.now();
+        dt = currentTime - lastFrameTime;
+        const elapsedTime = currentTime - startTime;
+
+        if (elapsedTime >= 1000) {
+            calculatedFPS = Math.floor((frameCount / elapsedTime) * 500);
```
Comment (ArjixWasTaken) [inline raylib.js:121]: Something tells me that `* 500` is wrong, but when I initially used `* 1000` it was double the fps...
I haven't really done the math for this.

Comment (ArjixWasTaken) [reply raylib.js:121]: You can completely remove this `if` statement along with the related variables if you don't plan on measuring the fps.

**Comment by CoolCat467** [comment]
I think it would be much better to separate the physics calculations from re-drawing frames.

My reasoning for this is that re-drawing frames with `requestAnimationFrame` means we are not wasting processing power re-drawing canvas that is not even visible to user when tab is in the background.
Additionally, we might encounter some quite interesting issues that in effect would be similar to screen tearing if (as you have noted) more than one frame is being drawn at once using your proposed solution.

There are two downsides to this approach I can think of immediately however. 
First being that it's more of a burden to program when what controls how object is displayed vs movement etc is separated.
Secondly, we might start seeing interesting things happen if frame is being drawn at the same time physics calculations are in the middle of changing something, for example multiple "sprites" are moving as one and physics calculation is only partially done, we will observe an effect akin to vibrating with the individual sprites.

Potential mitigations to this would be for physics and rendering to call the same function but have function take a boolean argument to denote if calculating physics or rendering.
To fix vibrating effect, that would be more tricky, but possible solution could be using something like a lock for when physics are being calculated and skip rendering new frame until physics is done.

**Comment by ArjixWasTaken** [comment]
> Additionally, we might encounter some quite interesting issues that in effect would be similar to screen tearing if (as you have noted) more than one frame is being drawn at once using your proposed solution.

No, that is not what is happening and is not what I noted.

I am using `setTimeout`, it won't re-run until the end of the frame, because it is invoked again at the end of the frame.
`setInterval` is what I noted would be problematic.

**Comment by CoolCat467** [comment]
Ah, I have misunderstood, thank you for correcting me.

**Comment by rexim** [comment]
The code have changed significantly already, sorry. I strongly recommend to just make another PR if you still want to contribute this. Thank you for all your contributions! :+1:

**Pull request closed without merging**